src/config/builder.rs

```diff
@@ -0,0 +1,150 @@
+use super::{
+    compiler, default_data_dir, Config, GlobalOptions, SinkConfig, SinkOuter, SourceConfig,
+    TestDefinition, TransformConfig, TransformOuter,
+};
+use crate::event;
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, Default)]
+#[serde(deny_unknown_fields)]
+pub struct ConfigBuilder {
+    #[serde(flatten)]
+    pub global: GlobalOptions,
+    #[serde(default)]
+    pub sources: IndexMap<String, Box<dyn SourceConfig>>,
+    #[serde(default)]
+    pub sinks: IndexMap<String, SinkOuter>,
+    #[serde(default)]
+    pub transforms: IndexMap<String, TransformOuter>,
+    #[serde(default)]
+    pub tests: Vec<TestDefinition>,
+}
+
+impl Clone for ConfigBuilder {
+    fn clone(&self) -> Self {
```

Comment:
We talked about adopting bincode (https://github.com/servo/bincode/blob/master/Cargo.toml) or cbor (https://github.com/pyfisch/cbor), maybe this is a good chance for that?

Reply:
It could be, but I'm not sure the benefit of changing it at this point. JSON works fine and this is not a performance sensitive code path.